engine/src/cmd/collision.cpp

```diff
@@ -156,96 +141,104 @@ void Collision::shouldApplyForceAndDealDamage(Unit* other_unit)
-        elasticity = std::max(elasticity, 0.0);
-        elasticity = std::min(elasticity, 1.0);
+        return;
     }
+    
+    double magnitude = new_velocity.Magnitude();
+    QVector velocity_norm = new_velocity.Normalize(); // want to perform positional displacement in direction of new velocity vector
+    double movementKludgeMagnitude = 2 * (new_angular_velocity.Dot(velocity_norm) * unit->radial_size / PI + magnitude);
+    QVector movementKludgeAmount = movementKludgeMagnitude*velocity_norm + abs(movementKludgeMagnitude)*normal;
+    unit->SetPosition(unit->Position() + (movementKludgeAmount)*SIMULATION_ATOM); // move by one physics frame at F() resultant linear velocity + component of angular velocity along linear velocity vector + normal before applying force -- this is NOT REMOTELY correct, but will often be sufficient
+}
 
+// apply force and torque, enforce clamping
+void Collision::applyForce(QVector &force, QVector &location_local)
+{
     if(!apply_force)
     {
         return;
     }
 
-    // Shorten variables for better visibility
-    float& m1 = mass;
-    Vector& v1 = velocity;
+    /*
+    Disabling clamping for the time being - will revisit this later; will be somewhat cleaner once moment of inertia implementation is upgraded.
+    May also want to do clamping in a configurable fashion AI vs. player, etc.
+
+    //Collision force caps primarily for AI-AI collisions. Once the AIs get a real collision avoidance system, we can
+    // turn damage for AI-AI collisions back on, and then we can remove these caps.
+    //value, in seconds of desired maximum recovery time
 
-    // Perfectly elastic result
-    Vector new_velocity = v1*(m1-m2)/(m1+m2) + v2*2*m2/(m1+m2);
 
-    // Apply ratio
-    new_velocity = new_velocity * elasticity;
 
-    // Apply force
-    unit->ApplyForce(new_velocity);
+    //scale max applicable force by unit's ability to restore orientation and velocity, then convert from frames to seconds
+    float  max_force = max_force_multiplier * (unit->limits.forward + unit->limits.retro + unit->limits.lateral + unit->limits.vertical) / (unit->sim_atom_multiplier * SIMULATION_ATOM);
+    float  max_torque = max_torque_multiplier * (unit->limits.yaw + unit->limits.pitch + unit->limits.roll) / (unit->sim_atom_multiplier * SIMULATION_ATOM);
+    float force_magnitude = force.Magnitude();
+    if (force_magnitude > max_force) {
+        force *= (max_force / force_magnitude);
+    }
+    if (torque_magnitude > max_torque) {
+        torque *= (max_torque / torque_magnitude);
+    }
+    
+    */
+
+    // Apply force at location, should just be one call to ApplyLocalTorque.... but there's a catch - our current moment of inertia units(as in SI units, not unit class) are a bit b0rked, so we need to do a workaround :-(
+    Vector torqueForce = force * (1 / (0.667 * unit->radial_size * unit->radial_size)); // use shell approximation
```

Comment:
same for the 0.667

Reply:
possibly the `1` as well

Reply:
The 1 is just an inversion - I'd leave the 1 as is.

Reply:
> same for the 0.667

Agreed -  the 0.667, as well as the need to access radial_size is a bandaid over a longer-term fix; it will eventually go away when we renovate the moment of inertia to not be scalar in the first place and also fix the dataset accordingly to specify the proper 3x3 matrices.